File: update.py
import os
import sys

os.system('go clean -cache')
os.system('go clean -modcache')

hasTest = False
hasNoTidy = False 

if len(sys.argv) >= 2:
    hasTest = sys.argv[1] == 't'
    hasNoTidy = sys.argv[1] == 'n'

if len(sys.argv) >= 3:
    hasTest = hasTest or sys.argv[2] == 't'
    hasNoTidy = hasNoTidy or sys.argv[2] == 'n'

cmd = 'go get'
if hasTest:
    cmd = cmd + ' -t'
cmd = cmd + ' -u ./...'
os.system(cmd)

if not hasNoTidy:
    os.system('go mod tidy')

File: update.lua
require "os"

os.execute('go clean -cache')
os.execute('go clean -modcache')

local hasTest, doTidy = false, true

if #arg >= 2 then
  hasTest, doTidy = arg[1] == 't', arg[1] ~= 'n'
end

if #arg >= 3 then
  hasTest = hasTest or arg[2] == 't'
  doTidy = doTidy and arg[2] ~= 'n'
end

local cmd = 'go get'
if hasTest then
  cmd = cmd .. '-t'
end
cmd = cmd .. ' -u ./...'
os.execute(cmd)

if doTidy then
    os.execute('go mod tidy')
end